### PR TITLE
CereStation

### PR DESCRIPTION
**Pull request by S34NW**
## What Does This PR Do
Ports CereStation, an out-of-rotation map from TG. It is a larger map (28k non-space turfs, vs Box's 17k, with about 3k more "playable" aired turfs, 15k vs 12k) that has each major department on its own asteroid.
The map is relatively dimly lit, with a focus on feelings of isolation and loneliness. While box gets cramped at 70+ users, this map will maintain its atmosphere, even on higher pops.
### Key map features:

- Every department is separated into its own asteroid island inter-connected by bridges.
- Every department has its own separate SMES storage that departments feed off of for power. Engineering has several wires that lead out leading to these departmental SMES'.
- Disposals is mostly external rooms jutting out of the asteroids, flinging trash/packages/corpses through several launch pads in space. Bring some coffee and a set of internals if you plan on surviving the disposal ride.
- Engineering and Atmospherics get several compact pickaxes to work around the asteroid rock, with emergency picks scattered around maintenance for anyone who wants to dig out their own assistant fort/body storage.
- All heads spawn on the bridge in their own personal quarters instead of in their department office, including the QM (as the QM is not a real head, they don't have main bridge access, only foyer access). Meet your fellow heads of staff before dying to revs/cult/incompetence.
- Brig is set up in a cell-block fashion, with the Warden sitting next to the main entrance to the cell block.
- Most APC's are in maintenance, allowing sneaky/really obvious sabotage. This balances the more "defended" elements of some departments. AI can still see them through the walls via cams, so they can interact with 90% of them as normal.
- Security has departmental offices, with "Criminal delivery chutes" than run on a secondary (and safe to use) disposals system, to deposit criminal scum into the brig lobby for processing. This is important due to the size of the map and para's move speed.
- Quantum pad network connecting the outer corner asteroids, these ~~need to be set up~~ now work at roundstart and work very poorly with stock parts, meaning as the round progresses more use of the quantum pads will be seen.
- Asteroids have the "Asteroid sand" base turf, as opposed to space. This means when the floor tiles are deconstructed (usually explosively) they won't create space tiles and plating, but space and asteroid sand. Non-asteroid areas still turn into space as usual.

### Things that likely should be done in the future:

- Medbay could use a remap, I've patched it up to be functional, but as I am not a medical player, I am not the best person to look at this area.
- ~~Brig could be remapped, it is currently quite inefficient in its use of space, while this does contribute to the map aesthetic, functionality should take precedent above form.~~
- The map will need extensive testing due to the nature of porting maps. While I have done my best to catch all the issues and bring it up to paradise standards, I will have missed a few things.
- ~~Since map rotation is not in yet,~~ this map will need to be standardised with any upcoming changes to box that get merged before rotation makes this available.
- ~~Add more hardsuits and softsuits on the various departments, since this _is_ a space-focused map.~~
- ~~Look at escape pod placement.~~
- ~~Needs a psych office~~
- ~~Needs a paramedic office~~

## Why It's Good For The Game
We currently lack a map that was designed for a higher population. Box was created with <70 people in mind, and while it has been adapted by us to fit our needs, it is still cramped.
Our other potential highpop maps; meta and delta, are severely outdated and need extensive work. While this work is being done, starting from scratch is easier, and having additional maps in rotation will add to variety. The benefits of having this map are, in my opinion, well worth the additional maintenance cost.

### Detailed view of the map:
(click to expand)
cere-min

### Size comparison vs. Box
ezgif com-gif-maker

## Changelog
:cl:
add: Added CereStation, a new highpop map.
/:cl:

### CODE REVIEW

**Comment by hal9000PR** [comment]
COWABUNGA. THE MAD MAN DID IT

**Comment by S34NW** [comment]
### _I can hear the bug reports rumbling on the hills, but:_
image

File: _maps/cere.dm [hunk 1]
```diff
@@ -0,0 +1,22 @@
+/*
+Original design by MMMiracles of TG Station
+Ported by S34N to Paradise
+*/
```
Comment (AffectedArc07) [inline _maps/cere.dm:4]: Expect this entire file to be redone once #12345 is merged

File: code/modules/map_fluff/cerestation.dm [hunk 2]
```diff
@@ -0,0 +1,10 @@
+/datum/map/cerestation
+	name = "CereStation"
+	full_name = "NSS Cere"
```
Comment (AffectedArc07) [inline code/modules/map_fluff/cerestation.dm:3]: Same with this. Also needs a shorthand name thats not the map name as is the fashion. @Necaladun may have input here.

**Comment by AffectedArc07** [comment]
Let the first round of nitpicks begin

image

Arrivals quantum pad room: Shouldnt these be reinforced glass? Same applies for all window exteriors.
image

Escape needs **way** more chairs:
image

Medbay is way too vacant but this needs a remap anyways as said above
image

This whole corridoor for the mining shuttle feels surplus when you can just flip the docking port
image
See: Deltastation
image

Tcomms room looks really off. Possibly take a leaf out the metastation or deltastation AI sats
image
image
image

Remove this other bed from the captains office. Its only going to encourage bad things
image

Same with all these dorms rooms. iirc we removed dorms for a reason, and I dont like the reason. Possibnly convert into an arcade.
image

Gamma armoury only being accessible through execution feels off
image

Give the vault dark flooring for extra aesthetic
image

Stick a mr changs vendor in here
image

We dont use the auxiliary mining base on this code. replace with asteroid wall or a regular construction area.
image

Unsure if these are disaster counters or just monitors, and if the latter, they need adding.
image

What is this shuttle + shuttle console for
image

Atmos needs more hardsuits
image

All out-of-room APCs could do with being moved inside
image

**Comment by Vi3trice** [comment]
Landmarks in general will need a go over for the various spawns, but most notably, arrivals has no arrivals. Latejoiners spawn into nothing.

CereLandmark

**Comment by AffectedArc07** [comment]
Approving of this as a concept but will put my final approval down once its up to par.

**Comment by S34NW** [comment]
Thanks for the help, the things you've mentioned have been fixed.

> We dont use the auxiliary mining base on this code. replace with an asteroid wall or a regular construction area.
Indeed, the console and mining stuff are gone now. It was meant to be a regular construction area.

> What is this shuttle + shuttle console for
It's just fluff for the engi sat, an abandoned shuttle seemed relevant. The console was not meant to be functional, this has been fixed.

> All out-of-room APCs could do with being moved inside
I feel this is a map specific thing that fits the style, however I won't object to moving them if there are issues with having them outside of the physical room (area tiles cover them, so they still work as normal for that room). With so much of this map walled in by asteroid, it gives antags an alternative method of break in: de-power the doors and crowbar their way in.

**Comment by Fox-McCloud** [comment]
Our present two maps aren't even up to standard yet nor are they ready for prime-time.

We have a small dev team that has a lot of burdens already with maintaining multiple maps---heck, TG only has 2 more maps than we do and their dev team is gigantic.

I'd rather not add more to this burden when our affairs are *already not in order*, not to mention making the turf transition even *more* of a nightmare to handle.

There' also the notion of a map of this size entrenching the idea that 150 or permanent ultra high population status is a good/admirable thing that we should aim for. I'm fundamentally opposed to this, as I firmly believe this will trample RP standards...as has been demonstrated numerous times in the past.

And, of course, there's the performance considerations---both memory and CPU. Larger maps inherently are going to use more memory--while that's not a major problem *yet*, we're only a few major features away from having to given that serious thought and consideration.

And, of course, a map with a lot more turfs is just a map with a lot more active turfs--a room twice the size is going to have twice the impact when a window breaks/someone breathes/what else have you.

So for reasons for performance, memory, not to mention maintainability (and near-future refactors), I'm going to have to object to this.

Our house is far from in order, and I'd rather not add a gigantic additional factor to maintain, on top of it.

**Comment by S34NW** [comment]
This map can't cope with 150 either, not optimally (What map even can?).
Even if (when) a server split happens, we still need "highpop" maps because 70+ is highpop. Delta has solid design, Box with all our tweaks is good enough for a range of pops.
But meta is so similar in box to design, why do we even want it? If maintenance cost is such a pressing matter, ditch it for a map that does something unique. With what you were saying about playercount taking up most of our CPU, having a large map should be okay with a <100 player count.

**Comment by Kiyahitayika** [comment]
The main corridors need vents, scrubbers, and air alarms and would need to be sectioned into separate areas for atmos to fill them properly (one air alarm per area). If I'm not mistaken, right now, the corridors with insulated floors would require folks to be wearing hardsuits, masks, and gas tanks to traverse.

**Comment by warior4356** [comment]
I personally I agree with almost all of Affected's feedback on this map, and I also applaud how quickly you've made the desired changes. The only point I disagree with is the out-of-room APCs. I like the antag opportunities this opens up. Though it does also open things up for griefing, so I'm not totally sure on this one.

**Comment by S34NW** [comment]
> 
> 
> The main corridors need vents, scrubbers, and air alarms and would need to be sectioned into separate areas for atmos to fill them properly (one air alarm per area). If I'm not mistaken, right now, the corridors with insulated floors would require folks to be wearing hardsuits, masks, and gas tanks to traverse.

Good point, they would still have air at roundstart but no way of replenishing it. I'll make the changes soon.

**Comment by warior4356** [comment]
An issue I think you missed. I only count 2 escape pods.

**Review by Kyep — COMMENTED** [review]
While I am generally opposed to non-box maps, especially ones from TG, I have to admit this map looks pretty nice. Perhaps that's because I haven't actually played on it... but it certainly looks more appealing than metastation (though equally, that's a very low bar to pass).

I'm not voting on this yet, but I'd like to see it testmerged a bit before I decide how to vote.

**Comment by Dumbdumn5** [comment]
As soon as this is up to scratch I'll put an approval comment in here, but can I say

Holy fucking shit that looks absolutely kickass. That's actually really fucking cool and I really, really like the concept. :+1:

**Comment by notfrying1pans** [comment]
oh no

**Comment by S34NW** [comment]
Due to the unique cargo dock layout in Cere, I've had to edit the supply shuttle to have two sets of doors. This means an edit to CC, I also put down an atmos blocker fan on the dock so shuttles don't get spaced.

**Comment by TheYeetster1** [comment]
Look how cute beepsky is
cute

**Comment by Sirryan2002** [comment]
I thoroughly enjoyed running through the map, I only found a few issues.

**A locker and punching bag spawn on the same tile in the dorms area**
possiblemapissue

**Fore Asteroid Solars has an airlock and Rwall on the same tile**
ForeAsteroidSolarsIssue

**I did not see the tcomms/monitoring codes spawn anywhere on station**

**Comment by Freakycrafter** [comment]
The quantum pads seem to start unlinked.
image
Not sure if this was intentional or not though.

**Comment by QuacksQ** [comment]
In the current state of this map (when it was test merged) atmos has no gasses so it is completely useless. You cannot make a teg, an SM have gas for the distro or backup n2 for when the SM goes bad.

In addition to that The air holding tank for the actual air supply itself has an issue where the tank moniter console says: Tank cannot be found This is due to an issue where the gas sensor in the tank is on the wrong frequency then the console itself

**Comment by Freakycrafter** [comment]
One of the windoors in front of the cap's office seems to be tinted or something?
image
I don't think it's supposed to be, is it? And if it were, why only that one windoor?

**Comment by Kael2244** [comment]
imagen
imagen

Enlage this corridor to a 3 tile wide , please.

**Comment by thespy24574** [comment]
No clown or mime space suit in EVA

**Comment by Kael2244** [comment]
imagen
The scape is way too small for an above 100 players map
imagen
Make a second entrace for evac so the traffic is more fluid please.

**Comment by Kael2244** [comment]
Explosive ores can spawn in the steroids making an INCREDIBLE mess , not sure if intended but that should be removed for the shake of the server.Or not (?)

**Comment by Kael2244** [comment]
imagen
Suit storage is missing walls on the left

**Comment by Kael2244** [comment]
imagen
Chemistry lab is missing the reinforced walls.

**Comment by Sirryan2002** [comment]
Security Airlocks at (185,132) and (191,132) have different permissions. The  (191,132) airlock is correct.

**Comment by warriorstar-orion** [comment]
Cargo shuttle corners are inverted.

Paradise Station 13-16_43_52

**Comment by AffectedArc07** [comment]
image

Active turfs after a normal traitor round. So much of the station becomes active on this map when even a minor piece of damage happens due to the map having large rooms that can all start reacting. That and people instantly breach the asteroid and make gas differences because "I am digging"

Considering this is a map designed for high pop, and we were at 120% CPU **at roundstart** due to said high pop. You blend in the atmospheric issues and you get a server that really starts to slow down. Movement is noticeably, and ingame events take longer. A 2 minute shuttle movement took 3 IRL minutes due to the extra delay. 

I can't support this if its going to put the server on its knees this quickly and to this level.

**Comment by shmabuu** [comment]
Well, performance on this map was atrocious. Only thing I can say that hasn't been said is that perma is missing the tools that are present on box being the nutrient jugs.

**Comment by Wojz** [comment]
Garden is missing a biogenerator and seed extractor. Also maints needs to be broken up into rooms so one missed pickaxe swing does not fuck everything up. And perhaps there shouldn't be maint pickaxes at all, since it's easy to breach on accident. Leave it to people with mesons. 

Also removing ores from the asteroids would again cut down on people digging through shit (and finding gibtonite).

**Comment by S34NW** [comment]
> 
> 
> image
> 
> Active turfs after a normal traitor round. So much of the station becomes active on this map when even a minor piece of damage happens due to the map having large rooms that can all start reacting. That and people instantly breach the asteroid and make gas differences because "I am digging"
> 
> Considering this is a map designed for high pop, and we were at 120% CPU **at roundstart** due to said high pop. You blend in the atmospheric issues and you get a server that really starts to slow down. Movement is noticeably, and ingame events take longer. A 2 minute shuttle movement took 3 IRL minutes due to the extra delay.
> 
> I can't support this if its going to put the server on its knees this quickly and to this level.
> 
> Heres some CPU stats on what this means.
> 
> On a normal round, the two most intensive subsystems are the atmospherics subsystem (SSair), and the key control system (SSinput). Here are some CPU stats on them from a normal round. Atmos is the main one to look at here, input is just for comparison sake.
> 
> ```
> /datum/controller/subsystem/air/fire - 466.848 seconds of CPU time in total (23513 calls) - 0.01s per call on average
> /datum/controller/subsystem/input/fire - 381.934 seconds of CPU time in total (163108 calls) - 0.0023s per call on average
> ```
> 
> Now heres the one from the cere testmerge
> 
> ```
> /datum/controller/subsystem/air/fire - 1032.24 seconds of CPU time in total (48197 calls) - 0.021s per call on average
> /datum/controller/subsystem/input/fire - 515.329 seconds of CPU time in total (96051 calls) - 0.005s per call on average
> ```
> 
> Total call count **does include resumes**, hence why the atmos call count on cere is higher. But you also have to keep in mind that it is taking double the time to process everything on average, and remember, this is in a shorter round than the normal 2 hours.

This is a major problem really, because if it wasn't just a one-off then it will just be the end of this map.
Were the statistics used in comparison from a similarly (160+) highpop round on box with an SM delam?

As much as I'd like to blame it on the crazy pop numbers, it might be a core issue. I think a test during quieter times (with closer to 120 online) might give some valuable insights.

**Comment by AffectedArc07** [comment]
> > 
> 
> > 
> 
> > image
> 
> > 
> 
> > Active turfs after a normal traitor round. So much of the station becomes active on this map when even a minor piece of damage happens due to the map having large rooms that can all start reacting. That and people instantly breach the asteroid and make gas differences because "I am digging"
> 
> > 
> 
> > Considering this is a map designed for high pop, and we were at 120% CPU **at roundstart** due to said high pop. You blend in the atmospheric issues and you get a server that really starts to slow down. Movement is noticeably, and ingame events take longer. A 2 minute shuttle movement took 3 IRL minutes due to the extra delay.
> 
> > 
> 
> > I can't support this if its going to put the server on its knees this quickly and to this level.
> 
> > 
> 
> > Heres some CPU stats on what this means.
> 
> > 
> 
> > On a normal round, the two most intensive subsystems are the atmospherics subsystem (SSair), and the key control system (SSinput). Here are some CPU stats on them from a normal round. Atmos is the main one to look at here, input is just for comparison sake.
> 
> > 
> 
> > ```
> 
> > /datum/controller/subsystem/air/fire - 466.848 seconds of CPU time in total (23513 calls) - 0.01s per call on average
> 
> > /datum/controller/subsystem/input/fire - 381.934 seconds of CPU time in total (163108 calls) - 0.0023s per call on average
> 
> > ```
> 
> > 
> 
> > Now heres the one from the cere testmerge
> 
> > 
> 
> > ```
> 
> > /datum/controller/subsystem/air/fire - 1032.24 seconds of CPU time in total (48197 calls) - 0.021s per call on average
> 
> > /datum/controller/subsystem/input/fire - 515.329 seconds of CPU time in total (96051 calls) - 0.005s per call on average
> 
> > ```
> 
> > 
> 
> > Total call count **does include resumes**, hence why the atmos call count on cere is higher. But you also have to keep in mind that it is taking double the time to process everything on average, and remember, this is in a shorter round than the normal 2 hours.
> 
> 
> 
> This is a major problem really, because if it wasn't just a one-off then it will just be the end of this map.
> 
> Were the statistics used in comparison from a similarly (160+) highpop round on box with an SM delam?
> 
> 
> 
> As much as I'd like to blame it on the crazy pop numbers, it might be a core issue. I think a test during quieter times (with closer to 120 online) might give some valuable insights.

The SM delam was in the last 10 minutes. Atmos was having a shit fit wayy before that. Player scaling doesn't scale atmos so that wouldn't increase the time for atmos operations. Population can be to blame for the SSinput increase, but not SSair.

**Comment by S34NW** [comment]
So it's due to room size?

**Comment by Wojz** [comment]
Box rooms are much smaller so a change in atmos activates a smaller area in my understanding.

**Comment by Englishformula** [comment]
- Missing toolbelts in RnD (normally 2)
- RD Office Missing advanced welding goggles
- Science is missing a sort of break-room, but vending machines will do. 
- Robotics is cluttered, but this is a nitpick. 
- Entrance to Science is not Reinforced Walls.
- Currently the RnD setup is on the opposite side of  the room from where the Front-desk is. It should probably be on  the same side as the front-desk window.

This is everything I have found related to Science Department.

**Comment by Rurik123** [comment]
Was suggested I put my thoughts on the sec layout, from the brief time I played it, here:

- In general, sec layout feels odd and inefficient, as stated in the PR description, a lot of wasted open space.
- Perma needs to have some sort of internal escape method (even if it spacing yourself and most likely dying) like box station, assuming we want to keep that same dynamic. Exposing it to space with a few reinforced windows seems the simplest solution. 
- Cells need to be external, not internal, preferably with the same airlock method that lets the criminals walk out after their timer is served. Forcing officers to have to let each individual petty criminal out will bog down sec during peak hours.
- Armory needs a few windows so officers can peer inside, like box station. With its current set up, any would be robber is almost completely sealed from view while pilfering the place. In addition to this, a camera outside armory space would be a good add.
- The wall chargers/locker room feels a bit too separated from everything else, but that might just be me. If there's anything that should be in the heart of security, it should be the wall rechargers.

Almost feels as if HoS/Warden's office, locker room, processing and brig bay should be moved upward while perma and the holding cells should be shifted downward, IAA/Magi's office being tossed across the hall. But, all that sounds like a pain in the ass so I don't blame anyone for not wanting to deal with that.

On a side note, what in gods name is that disposals system.

**Comment by hal9000PR** [comment]
The disposals system is beautiful.

I might not be the first to say this but departures needs to be waaaayy bigger.

**Comment by Christasmurf** [comment]
-First thing to note, no lights in the arrivals part to the right of the arrivals shuttle 
image
-Secondly, a bunch of maintenance areas have wires going directly over asteroid sand which means if a mouse eats them either you gotta slap a tile down to replace them or reroute the things. You may not want to change this I'm just thinking it could cause problems because it took me a few minutes to figure it out.
-Thirdly, and this may be related to the asteroid sand thing, pretty much none of the APCs in some areas are charging. Not sure why, because the wires look fine and the SMES are outputting. Not sure if the asteroid sand with wires on are unable to transfer energy? I had to set up solars to get certain areas of the map to get power. (For example, on my current test, I have science solars and supermatter active. All of science and service has power, only after I turned the solars on; cargo, brig, command and engineering have power, however medical is limited to virology, the big surgery room, the psyche office and the public hallways. The rest of medbay is unpowered with no broken wires between any of those APCs and the powered areas.) This issue may be caused by my incompetence though.

**Comment by S34NW** [comment]
@Christasmurf could you DM me on discord about the power issue so I can investigate further? The departmental solars should only hook into that asteroids power grid, not the wider grid. So it may be a SMES issue.

**Comment by ppi13** [comment]
Due to some issues with the map, most astroid turf looks like space, so ignore that.
This here is a bit too small, and a bit too far away, and is missing a carp spawnpoint for the bomb to bounce into and be sent back, or even be in range. Try keeping it around the same size and distance as it is on cyberiad,
image
What decided where apcs should be, research directors office, server room and toxin mixing is in maint, but server cold room and research lab, seems a bit inconsistent
image
Here it is again with resarch testing chamber and experimentor chamber being inside, but research testing lab being outside in maints, also experimentor chamber kinda close to research testing chamber that tends to be exploded, so moving research testing chamber to be hanging into space, could be a good thing.
image
More apc inconsistancy, having two tables and two windoors seems superfluous, having the vents and scrubbers closer to the machines would be good for ventcrawler reasons, and maybe move the tables in the middle, maybe move firestart e-n into rnd
image
to something like this, machinery close to each other, short way to tables, with this you can put the shutter button closer, as well as move the science request console up one tile, and maybe put the light fixture on the north wall instead. also the guest pass terminal is missing along with the copier, gotta have the copier so we can emag it for people to print their ass on.
image
swapping around the place of the computers and recycler/fridge/processor, ya wanna always be in range of your slimes, so they will be friendly to you incase they get angry and try to escape, the kill chamber could make do with being a seperate room, or you could convert it into the typical setups.
image
So turning the killchamber into something like this:
image
leaving 60 of the 40 needed cable on the solars, yes please
image
Robotics needs to be compressed down, you you should be able to see them and talk to them from the window, making em more unavailable and harder to contact without having to resort to radio is a denada for me, oil tank needs to be removed, maybe see if you can git in one or two more cyborg rechargers in mechbay as well, and 
image
This here is good, no looping atmospherics, makes it more dangerous when pipes are missing.
image
There are over a 100 decals here, a better alternative would be to sprite floortiles with these decals on them, then to keep this many. Don't forget how frees remap added like 600 new decals, which seemed to be enough for many players to have some performance issues.
image
On the topic of decals, maints got a pointless amount of mouse traps, it would be better to remove em, and just get down the object count

More decals, also might be good idea to move sec pod officer and pod to brig area, that way the mechanic can keep doing traitorous things without fear of the pod pilot pooping up, and for arrival for the same reasons.
image
Avoid scrubbers and vents ontop of pipes
image
There are nearly 300 decals in this here, which should just be removed, straight up, keep the cobwebs, maybe add a few extra, but all that dirt, along with the delivery decals is just pointless use of effects, to put into contesxt, there are 344 `/obj/effect/decal/cleanable/dirt` on the map and nearly 300 of them are here, and there are 119 more cleanable subtypes, which should probably be cleaned up, maybe keep it to mostly abandoned rooms at that
image
Atmospherics is a thing, but whats the deal with all those windows and windoors, best to remove as we wanna limit those types of windows, and there are no vents and scrubbers in atmos storage, and instead of the delivery turf decal, use the color approriate warning stripes hollow instead. also, vent ontop of scrubber pipe
image
Unsure how i feel about the southern CE office windows, and i feel like it could be good with an atmos door into engineering, i dunno really
image
Could you add two more emitters here, along with removing those two space cameras, ai don't need to see that shit
image
vents ontop of pipes
image
Crematorium missing a freezer, and the switch for the incinerator is awkwardly placed.
image
Cryo is a bit small, if you can make it bigger, you can avoid scrubber ontop of pipe
image
Botany and kitchen should preferable be wall to wall, and kitchen is weirdly small towards the south, widdening the kitchen and adding more tables, anno cyberiad, due to the fact that so few uses the bar for food, thats the main access to food
image
This room here is too dirty, there are nearly 50 cleanable effect in a 5 by 6 room
image
No bonus wires at these solars???
image
But these neighbouring ones Do got bonus wires
image
5 cargo consoles? 4 mules bots? Might wanna rotate the cargo shuttle here, and consolidate the space, and now it's setup so baldies can't space crate, oh and should there not be more spawnpoints for miners???
image
ugh, ai sat looks so uneven down here, and there is no microwave with donkpockets nearby, and it's weirdly gross how the rooms are
image
Man, this is some good piping, even if it makes it a pain for vent crawlers, a+++++++
image
Why so beeg, is it setup to work with tripple ai? Why the gross windows when we do not have diagonal windows, the walls/windows in the middle seems more of a detriment then a help.
image
Is there supposed to be no access on these doors??
image
Why is there a qm office at bridge?
image
How will i bridge hobo??? Also, having departmental lockdown available from bridge?
image
Hos gets the wrong gloves in his closet >=C
image
Ya kno, last time i saw a map where the captains quarters was exposed to space, even if there is a layer of astroid not showing up, this did change the winrate of nukies on that server from maybe, to basically yes, also why is there two showers, and
image
Removing the window tiles and booze machine and have it something like this, would probably be better, also where is the bob ross painting?
image
Arcade could probably shrink a bit, and look another cryo room??? with scrubbers on pipes, 
image
Okay, brig cells here are... confusing, door timers are weirdly placed, dunno how i feel about this open shared cell practice.
image
Why so many secways?
image
Courtroom being smaller is good, processing being public is also good
image
Brigphys kinda small, also could put a medivendplus there so they stop stealing one from medbay.
image
On the topic of medbay, i forgot ALL about it.
Welder tanks could be closer to the chemists
image
None of these weird window tiles are needed
image
Instead of having this big surgery room, and big viewer room, just put two surgery rooms instead.
image
Others things of note is that some offices got weird pets, and that there are like 3 sets of disposals, one for trash, one for delivery and one for security, there is a weird inconsistancy with apc in maint, or not in maint, devices ontop of pipes, and overuse of effects.

**Comment by WingedYordle** [comment]
As requested, have red tiles that sound the perma lobby and perma to indicate when an attempted escape is now an escape. Eliminate the confusion of are they in or are they out?

**Comment by S34NW** [comment]
> 
> 
> As requested, have red tiles that LINE the perma lobby and perma to indicate when an attempted escape is now an escape. Eliminate the confusion of are they in or are they out?

Yep! It's been idiot-proofed as much as possible.

**Comment by Spacemanspark** [comment]
Jesus fuck

Yes please, we need more fuckin' maps like this.
:+1:

**Comment by PacifistDalek** [comment]
I'm so hype, this looks a lot better than meta

**Comment by S34NW** [comment]
I've also made space-facing asteroid rock take 2x as long to mine and given it a custom name/description. This should make accidentally mining out into space (and thus depressurising everything) harder

**Comment by AffectedArc07** [comment]
Please modify `.github/workflows/ci.yml` and add the Cere datum to the CI list

**Comment by Wojz** [comment]
Bar doors are called "Kitchen" and are not common access.

Airlocks that the atmos pipes go through parallel to the glass corridors between asteroids seem to have common access, at least near bar.

**Comment by S34NW** [comment]
image
AT count on a 120 pop shift. Performance feels great, will supply metrics for perf soon.

**Comment by Wojz** [comment]
Merch vendor missing from near cargo, all external airlocks start unbolted and common access it seems.
No pet store.

Map ran super smooth despite pretty high-ish pop, there was absolutely no comparison to the previous TM. So much better. Extremely solid work from S34N upgrading it, reducing the distance between stuff, and just prettying it up with glass floors, colored carpets, railings, and such. Labor of love and it shows.

**Comment by JackoMallows** [comment]
IIRC the bridge... lounge/lobby(?) was accessible to anyone, not sure if that was intentional, but figured I'd mention it just in case. (Apologies if it turns out I misremembered as well, there were a lot of people around so I may have caught something else.) Everything else I noticed was already mentioned by Woj. 

Really great stuff S34N, I'd love to see this map in the standard map rotations in the future. Performance seemed perfectly stable, and it's clear a lot of work has gone into this, excellent job!

**Comment by PeppermintEclipse** [comment]
I'm not sure if this is a mapping-related issue or just a standard glitch, but was told to mention it here just in case; at one point someone trailed blood through the hall outside HoP and the stain on this specific tile couldn't be cleaned no matter what I did. The blood also layered above objects (but not people) instead of going underneath:
image

This room (I think it's Departures) has no lights in the southern half and is pitch-black in one corner as a result:
image

Also, the extra space and supplies in the janitor's closet are a _godsend_ - on any other map, having more than one janitor means immediately being undersupplied and cramped in the tiny closets - but it's worth pointing out that the janitor supply wardrobes each come with equipment for two people, so having two of them at roundstart feels a bit excessive and makes cargo orders a bit pointless (especially when there are only two of everything else). Since the biohazard wardrobe only comes with one bio suit, could I suggest replacing the supply wardrobe on the left with a second biohazard wardrobe?

**Comment by S34NW** [comment]
reopening for CI

**Comment by Fox-McCloud** [comment]
We have analyzed the performance metrics on this, and it's not good. Atmospherics is considerably more expensive (30%+) on Ceres, which is what we predicted. 

It initially seemed to perform equally to box, but it turns out someone did a lot of shenanigans with simple mobs which hugely bloated up the cost and put it in line with Ceres.

Performance alone is a huge reason this has been rejected.

At this point in time, I do not think we need a 4th map--especially when our affairs aren't in order for the present ones.

**Pull request closed without merging**